src/Morphir/Visual/ViewValue.elm

```diff
@@ -176,3 +194,62 @@ viewValueByLanguageFeature ctx argumentValues value =
                     ]
                     (XRayView.viewValue other)
                 ]
+
+
+parseArithmeticOperatorTree : ArithmeticOperatorTree -> String
+parseArithmeticOperatorTree arithmeticOperatorTree =
+    case arithmeticOperatorTree of
+        ArithmeticValueLeaf typedValue ->
+            helperFunctionValue typedValue
+
+        ArithmeticDivisionBranch arithmeticOperatorTree1 arithmeticOperatorTree2 ->
+            "ADB" ++ " -> Divide -> [ " ++ parseArithmeticOperatorTree arithmeticOperatorTree1 ++ " , " ++ parseArithmeticOperatorTree arithmeticOperatorTree2 ++ " ]"
+
+        ArithmeticOperatorBranch arithmeticOperator arithmeticOperatorTrees ->
+            case arithmeticOperator of
+                Add ->
+                    "AOB" ++ " -> Add -> [ " ++ String.join " , " (List.map parseArithmeticOperatorTree arithmeticOperatorTrees) ++ " ]"
+
+                Subtract ->
+                    "AOB" ++ " -> Subtract -> [ " ++ String.join " , " (List.map parseArithmeticOperatorTree arithmeticOperatorTrees) ++ " ]"
+
+                Multiply ->
+                    "AOB" ++ " -> Multiply -> [ " ++ String.join " , " (List.map parseArithmeticOperatorTree arithmeticOperatorTrees) ++ " ]"
+
+
+helperFunctionValue : Value ta va -> String
+helperFunctionValue value1 =
+    case value1 of
+        Literal _ literal ->
+            case literal of
+                BoolLiteral bool ->
+                    case bool of
+                        True ->
+                            "AVL (Bool) = True"
+
+                        False ->
+                            "AVL (Bool) = False"
+
+                CharLiteral char ->
+                    "AVL (char) = " ++ String.fromChar char
+
+                StringLiteral string ->
+                    "AVL (String) = " ++ string
+
+                IntLiteral int ->
+                    "AVL (Int) = " ++ String.fromInt int
+
+                FloatLiteral float ->
+                    "AVL (Float) = " ++ String.fromFloat float
+
+        Variable _ name ->
+            "AVL (Variable) -> " ++ Name.toTitleCase name
+
+        Reference va fQName ->
+            "AVL (Reference) -> " ++ FQName.toString fQName
+
+        Apply va value2 value3 ->
+            "AVL (Apply): " ++ helperFunctionValue value2 ++ " __ " ++ helperFunctionValue value3
+
+        _ ->
+            "Some other format"
```

Comment:
Do we need all this here?

Reply:
moved to to AbstractTreeVisualizer.elm in tests/Morphir